diff --git a/vendor/github.com/ipfs/go-ipfs/core/core.go b/vendor/github.com/ipfs/go-ipfs/core/core.go
index 0fa6b52cd..bed3010b6 100644
--- a/vendor/github.com/ipfs/go-ipfs/core/core.go
+++ b/vendor/github.com/ipfs/go-ipfs/core/core.go
@@ -453,7 +453,7 @@ func setupDiscoveryOption(d config.Discovery) DiscoveryOption {
 // HandlePeerFound attempts to connect to peer from `PeerInfo`, if it fails
 // logs a warning log.
 func (n *IpfsNode) HandlePeerFound(p pstore.PeerInfo) {
-	log.Warning("trying peer info: ", p)
+	log.Info("trying peer info: ", p)
 	ctx, cancel := context.WithTimeout(n.Context(), discoveryConnTimeout)
 	defer cancel()
 	if err := n.PeerHost.Connect(ctx, p); err != nil {
